web/client/components/map/leaflet/MeasurementSupport.jsx

```diff
@@ -2,21 +2,217 @@ const PropTypes = require('prop-types');
+            }
+            break;
+        case 'feet':
+            d *= 1.09361 * 3;
+            distanceStr = L.GeometryUtil.formattedNumber(d, p.ft) + ' ft';
+            break;
+        case 'nauticalMile':
+            d *= 0.53996;
+            distanceStr = L.GeometryUtil.formattedNumber(d / 1000, p.nm) + ' nm';
+            break;
+        case 'yards':
+        default:
+            d *= 1.09361;
+            if (options.distinctMeasure) {
+                if (distance > 1760) {
+                    distanceStr = L.GeometryUtil.formattedNumber(d / 1760, p.mi) + ' miles';
+                } else {
+                    distanceStr = L.GeometryUtil.formattedNumber(d, p.yd) + ' yd';
+                }
+            } else {
+                distanceStr = options.uom.length.unit === "mi" ? L.GeometryUtil.formattedNumber(d / 1760, p.mi) + ' miles' : distanceStr = L.GeometryUtil.formattedNumber(d, p.yd) + ' yd';
+            }
+            break;
+    }
+    return distanceStr;
+};
+
+/** @method readableArea(area, isMetric, precision): string
+ * @return a readable area string in yards or metric.
+ * The value will be rounded as defined by the precision option object.
+ * this override is necesary due to the customization on how the area measure label is presented
+ supporting also the square nautical miles and square feets
+ */
+L.GeometryUtil.readableArea = (area, isMetric, precision, options) => {
+    let areaStr;
+    let units;
+    let type;
+    const {unit} = options.uom.area;
+    let p = L.Util.extend({}, defaultPrecision, precision);
+    let a = area;
+    if (options.distinctMeasure) { // this is done for retrocompatibility
+        if (isMetric) {
+            units = ['ha', 'm', 'km']; // added km
+            type = typeof isMetric;
+            if (type === 'string') {
+                units = [isMetric];
+            } else if (type !== 'boolean') {
+                units = isMetric;
+            }
+            if (area >= 1000000 && units.indexOf('km') !== -1) {
+                areaStr = L.GeometryUtil.formattedNumber(area * 0.000001, p.km) + ' km²';
+            } else if (area >= 10000 && units.indexOf('ha') !== -1) {
+                areaStr = L.GeometryUtil.formattedNumber(area * 0.0001, p.ha) + ' ha';
+            } else {
+                areaStr = L.GeometryUtil.formattedNumber(area, p.m) + ' m²';
+            }
+        } else {
+            a /= 0.836127; // Square yards in 1 meter
+            if (a >= 3097600) { // 3097600 square yards in 1 square mile
+                areaStr = L.GeometryUtil.formattedNumber(a / 3097600, p.mi) + ' mi²';
+            } else if (a >= 4840) { // 4840 square yards in 1 acre
+                areaStr = L.GeometryUtil.formattedNumber(a / 4840, p.ac) + ' acres';
+            } else {
+                areaStr = L.GeometryUtil.formattedNumber(a, p.yd) + ' yd²';
+            }
+        }
+    } else {
+        if (unit === "sqkm") {
```

Comment:
Can we translate all these if(s) into a functional approach?

Reply:
indeed